### PR TITLE
Game fix

### PR DESCRIPTION
**Pull request by hopesgit**
Game refactor to fix some issues with the Game class and fixes for some edge cases

### CODE REVIEW

File: test/game_test.rb [hunk 1]
```diff
@@ -15,6 +16,74 @@ def test_it_has_attributes
...
+    game.stubs(:user_input_2).returns("6")
+    game.board_size_user_input
+
+    assert_equal 30, game.player.board.cells.count
+    assert_equal 30, game.cpu.board.cells.count
+  end
+
+  def test_player_placing_ships
+    game = Game.new
+    game.stubs(:user_input_1).returns("B1 B2 B3")
```
Comment (JesseMellinger) [inline test/game_test.rb:54]: Really like seeing the use of a stub here for user input.

File: test/game_test.rb [hunk 2]
```diff
@@ -15,6 +16,74 @@ def test_it_has_attributes
...
+    game.cpu.place_ship(game.cpu.cruiser, game.pick_random_ship_coordinates(game.cpu.cruiser))
+    game.stubs(:pick_random_ship_coordinates).returns(["C1", "C2"])
+    game.cpu.place_ship(game.cpu.submarine, game.pick_random_ship_coordinates(game.cpu.submarine))
+    game.stubs(:user_input_1).returns("B1 B2 B3")
+    game.stubs(:user_input_2).returns("C1 C2")
+    game.place_player_ships
+    game.stubs(:user_input_1).returns("C1")
+    game.user_get_coordinate_to_fire_on
+
+    assert_equal "H", game.cpu.board.cells["C1"].render
```
Comment (JesseMellinger) [inline test/game_test.rb:87]: Creative means of confirming ship has been hit and cell rendered correctly.
